### ISSUE TITLE
Gradient for Vector Outputs

### ISSUE THREAD

**Issue by gitvicky** [google/jax#2990]
Hey, 

Having constructed a Neural Network from scratch using JAX, is there anyway to obtain the gradient of each output with respect to each of the corresponding inputs. 

For example the predict function for the network been designed as (jax.numpy imported as  jnp): 

**def predict(params, inputs):
  activations = inputs
  for w, b in params[:-1]:
    outputs = jnp.dot(w, activations) + b
    activations = jnp.tanh(outputs)
  
  final_w, final_b = params[-1]
  field_output = jnp.dot(final_w, activations) + final_b
  return field_output**

Vectorised as : **batched_predict = vmap(predict, in_axes=(None, 0), out_axes=0)**

The function whose gradient needs to be taken is given as: 

**def func(X):
    return (batched_predict(params, X))**

Can we get the value : 
**grad(batched_predict)(X)**
and 
**grad(grad(batched_predict))(X)**

Effectively looking to estimate the higher order derivatives of the function output with respect to the input features given by the neural network at a point in the feature space

**Comment by shoyer** [comment]
Try [jaxfwd](https://jax.readthedocs.io/en/latest/jax.html#jax.jacfwd) and [jacrev](https://jax.readthedocs.io/en/latest/jax.html#jax.jacrev) for gradients of vector valued functions.

**Comment by mattjj** [comment]
I think Stephan answered this, so I'm going to close the issue now. Check out [the autodiff cookbook](https://colab.sandbox.google.com/github/google/jax/blob/master/docs/notebooks/autodiff_cookbook.ipynb) if you haven't already!
